test/integ/quote-relay.test.ts

```diff
@@ -0,0 +1,193 @@
+import chaiSubset from 'chai-subset';
+import _ from 'lodash';
+import { RoutingType } from '../../lib/constants';
+import { QuoteRequestBodyJSON, RelayQuoteDataJSON, RoutingConfigJSON } from '../../lib/entities';
+import { QuoteResponseJSON } from '../../lib/handlers/quote/handler';
+import { RelayOrderReactor__factory } from '../../lib/types/ext';
+import { getAmount } from '../utils/tokens';
+import { BaseIntegrationTestSuite, call } from './base.test';
+
+chai.use(chaiAsPromised);
+chai.use(chaiSubset);
+
+const SLIPPAGE = '5';
+
+describe('relayQuote', function () {
+  let baseTest: BaseIntegrationTestSuite;
+  let reactorAddress: string;
+
+  // Help with test flakiness by retrying.
+  this.retries(2);
+  this.timeout(40000);
+
+  let alice: SignerWithAddress;
+  let filler: SignerWithAddress;
+
+  beforeEach(async function () {
+    baseTest = new BaseIntegrationTestSuite();
+    [alice, filler] = await baseTest.before();
+    // deploy reactor
+    const factory = new RelayOrderReactor__factory(alice);
+    const reactorContract = await baseTest.deployContract(factory, [UNIVERSAL_ROUTER_ADDRESS(1)]);
+    reactorAddress = reactorContract.address;
+  });
+
+  for (const type of ['EXACT_INPUT', 'EXACT_OUTPUT']) {
+    describe(`${ID_TO_NETWORK_NAME(1)} ${type} 2xx`, () => {
+      describe(`+ Execute Swap`, () => {
+        it(`stable -> stable, gas token == input token`, async () => {
+          const quoteReq: QuoteRequestBodyJSON = {
+            requestId: 'id',
+            tokenIn: USDC_MAINNET.address,
+            tokenInChainId: 1,
+            tokenOut: USDT_MAINNET.address,
+            tokenOutChainId: 1,
+            amount: await getAmount(1, type, 'USDC', 'USDT', '100'),
+            type,
+            slippageTolerance: SLIPPAGE,
+            configs: [
+              {
+                routingType: RoutingType.RELAY,
+                protocols: ['V2', 'V3', 'MIXED'],
+                swapper: alice.address,
+                gasToken: USDC_MAINNET.address,
+              },
+            ] as RoutingConfigJSON[],
+          };
+
+          const response: AxiosResponse<QuoteResponseJSON> = await call(quoteReq);
+          const {
+            data: { quote },
+            status,
+          } = response;
+
+          expect(status).to.equal(200);
+          const order = RelayOrder.parse((quote as RelayQuoteDataJSON).encodedOrder, 1);
+          order.info.reactor = reactorAddress;
+
+          expect(order.info.swapper).to.equal(alice.address);
+          expect(order.info.input).to.not.be.undefined;
+          expect(order.info.fee).to.not.be.undefined;
+          expect(order.info.universalRouterCalldata).to.not.be.undefined;
+          expect(parseInt(order.info.input.amount.toString())).to.be.greaterThan(90000000);
+          expect(parseInt(order.info.input.amount.toString())).to.be.lessThan(110000000);
+
+          const { tokenInBefore, tokenInAfter, tokenOutBefore, tokenOutAfter } = await baseTest.executeRelaySwap(
```

Comment:
We wanted to avoid on-chain integ tests here as they tend to be flaky and slow - can we move these to SDK integ instead?

Reply:
yeah I had that thought too but I think we need tests here to test that the calldata for UR is being built correctly - there's not a great way to test that in the SDK level because the classic quote (which the calldata) is generated from is only available at the quote level within URA. Also, we don't mainnet fork for the SDK integ tests so we'd have to do that.

I kept these e2e tests simple and short - only two cases, and both doing stable to stable with reasonable bounds. I'll remove the checks that the input amounts are within an explicit range so the only remaining checks are relative